util/version/version.go

```diff
@@ -173,17 +173,57 @@ func (v buildIdentifier) compare(o buildIdentifier) int {
 	}
 }
 
-// CompareWithBuildIdentifiers compares 2 version a and b.
+// compareWithBuildIdentifiers compares 2 version a and b.
 // Perfoms a standard version compare between a and b. If the versions
 // are equal, build identifiers will be used to compare further.
 //   -1 == a is less than b.
 //   0 == a is equal to b.
 //   1 == a is greater than b.
-func CompareWithBuildIdentifiers(a semver.Version, b semver.Version) int {
+func compareWithBuildIdentifiers(a semver.Version, b semver.Version) int {
 	if comp := a.Compare(b); comp != 0 {
 		return comp
 	}
 	biA := newBuildIdentifiers(a.Build)
 	biB := newBuildIdentifiers(b.Build)
 	return biA.compare(biB)
 }
+
+type compare struct {
+	buildTags bool
+}
+
+// CompareOption is a configuration option for Compare.
+type CompareOption func(*compare)
+
+// WithBuildTags modifies the version comparison to also consider build tags
+// when comparing versions.
+// Perfoms a standard version compare between a and b. If the versions
+// are equal, build identifiers will be used to compare further.
+//   -1 == a is less than b.
+//   0 == a is equal to b.
+//   1 == a is greater than b.
+func WithBuildTags() CompareOption {
+	return func(c *compare) {
+		c.buildTags = true
+	}
+}
+
+func (c *compare) Compare(a, b semver.Version) int {
```

Comment:
is it possible to inline this in func Compater, thus avoiding one level of indirection